GoRogue/MapGeneration/Region.cs

```diff
@@ -12,139 +14,91 @@ namespace GoRogue.MapGeneration
-        public Point NorthEastCorner { get; private set; }
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(Lines.Algorithm algorithm, params Point[] corners)
+            : this(corners, algorithm, null) { }
 
         /// <summary>
-        /// All points within the region
+        /// Returns a new PolygonArea with corners at the provided points.
         /// </summary>
-        private MultiArea _points;
+        /// <param name="components">A component collection to use for this region</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(IComponentCollection? components = null, params Point[] corners)
+            : this(corners, Lines.Algorithm.DDA, components) { }
 
         /// <summary>
-        /// All of the boundary points of this region.
+        /// Returns a new PolygonArea with corners at the provided points.
         /// </summary>
-        public IReadOnlyArea OuterPoints => _outerPoints;
-        private MultiArea _outerPoints;
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="components">A component collection for this region</param>
+        /// <param name="corners">The points which are corners for this polygon</param>
+        public Region(Lines.Algorithm algorithm, IComponentCollection? components, params Point[] corners)
+            : this(corners, algorithm, components) { }
 
         /// <summary>
-        /// All of the points inside this region, excluding boundary points
+        /// Returns a new polygon with corners at the provided points, using the algorithm DDA to produce lines
         /// </summary>
-        public IReadOnlyArea InnerPoints => _innerPoints;
-        private Area _innerPoints = null!;
+        /// <param name="corners">The corners of the polygon</param>
+        public Region(params Point[] corners) : this(corners, Lines.Algorithm.DDA, null) { }
 
-        /// <summary>
-        /// All of the outer points along the southern boundary
-        /// </summary>
-        public IReadOnlyArea SouthBoundary => _southBoundary;
-        private Area _southBoundary;
-
-        /// <summary>
-        /// All of the outer points along the northern boundary
-        /// </summary>
-        public IReadOnlyArea NorthBoundary => _northBoundary;
-        private Area _northBoundary;
-
-        /// <summary>
-        /// All of the outer points along the eastern boundary
-        /// </summary>
-        public IReadOnlyArea EastBoundary => _eastBoundary;
-        private Area _eastBoundary;
+        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components = null)
+        {
+            Area = new PolygonArea(corners, algorithm);
+            GoRogueComponents = components ?? new ComponentCollection();
```

Comment:
Missing the portion from `RegionWithComponents` necessary to make parent-aware components work properly:

 ```suggestion
            GoRogueComponents = components ?? new ComponentCollection();
            GoRogueComponents.ParentForAddedComponents = this;
```